sql/codegen.go

```diff
@@ -225,307 +227,26 @@ func removeLastSemicolon(s string) string {
 	return s
 }
 
-func genTF(w io.Writer, pr *extendedSelect, fts fieldTypes, db *DB) error {
+func codegen(w io.Writer, pr *extendedSelect, fts fieldTypes, db *DB) error {
 	r, e := newFiller(pr, fts, db)
 	if e != nil {
 		return e
 	}
-	if e = codegenTemplate.Execute(w, r); e != nil {
-		return fmt.Errorf("genTF: failed executing template: %v", e)
-	}
-	return nil
-}
-
-const codegenTemplateText = `
-import os
-# Disable Tensorflow INFO and WARNING logs
-os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
-
-import sys, json
-import tensorflow as tf
-import numpy as np
-import functools
-try:
-    import sqlflow_models
-except:
-    pass
-
-from sqlflow_submitter.db import connect, insert_values, db_generator
 
-# Disable Tensorflow INFO and WARNING
-import logging
-tf.get_logger().setLevel(logging.ERROR)
-
-` +
-	// TODO(typhoonzero): get NUM_BUCKETS, EMBEDDING_WIDTH from Extended SQL statements in
-	// COLUMN sub clause
-	`
-BATCHSIZE = 1
-EPOCHS = None
-NUM_BUCKETS=160000
-EMBEDDING_WIDTH=128
-
-train_args = dict()
-{{range $key, $value := .Attrs}}
-{{if eq $key "BATCHSIZE"}}
-BATCHSIZE = {{$value}}
-{{else if eq $key "EPOCHS"}}
-EPOCHS = {{$value}}
-{{else}}
-train_args["{{$key}}"] = {{$value}}
-{{end}}
-{{end}}
-
-driver="{{.Driver}}"
-{{if ne .Database ""}}
-database="{{.Database}}"
-{{else}}
-database=None
-{{end}}
-
-conn = connect(driver, database, user="{{.User}}", password="{{.Password}}", host="{{.Host}}", port={{.Port}})
-
-{{$iskeras := .IsKerasModel}}
-
-feature_columns = dict()
-{{ range $target, $colsCode := .FeatureColumnsCode }}
-feature_columns["{{$target}}"] = []
-{{ range $col := $colsCode }}
-feature_columns["{{$target}}"].append({{$col}})
-{{ end }}
-{{ end }}
-
-
-feature_column_names = [{{range .X}}
-"{{.FeatureName}}",
-{{end}}]
-
-
-classifier = {{.Estimator}}(
-    **feature_columns,
-    **train_args,
-    {{if .IsKerasModel}}
-)
-    {{else}}
-    model_dir = "{{.Save}}")
-    {{end}}
+	estimatorKey := strings.ToUpper(r.Estimator)
+	// select estimatorTemplate
+	var temp estimatorTemplate
+	switch {
+	case strings.HasPrefix(estimatorKey, "TF.ESTIMATOR") || r.IsKerasModel:
```

Comment:
This would change users' habit, for the current implementation,  `.. TRAIN modelname ...` would train the [premade estimator mode](https://www.tensorflow.org/guide/premade_estimators), and `... TRAIN sqlflow_models.modelname` to train the [SQLFlow models](https://github.com/sql-machine-learning/models) which defined by Kears API.

Maybe you can create an issue first to discuss how to distinguish **TF Estimator Model**. ** SQLFlow custom model**, **XGBoost** and **ALPS**, after that we can implement this feature and update the user guide doc also.

In this PR, maybe we can use the prefix `XGBOOST` to train the XGBoost model, and TF model by default.

Reply:
Better to not use capitalized words in the model name, to fit the general SQL rules that not case sensitive.

Reply:
Currently, `ALPS` uses the environment variable `SQLFLOW_submitter` to determine which `codegen` to use, maybe we can also use this pattern here.

Reply:
@typhoonzero  
Maybe environment variable approach is not friendly to users? 
What's more,  I think `xgboost generator` can share a lot of information (such as `filler`) with default `TF generator`.  So, I try to split codegen into two stages.

Reply:
I think implement a common `Filler` struct is a wonderful idea, and maybe we need to think about it with  `AlpsFiller`, `TFFiller` and `XGBFiller`. 

And @sperlingxx can you separate this into two PRs:
1. Implement an independent `codegen_xgboost` like `codegen_alps`, and than
2. Refactoring the Filler struct.

Reply:
@sperlingxx 
> Maybe environment variable approach is not friendly to users?

Seems so, should let user to choose the model to train, for ALPS it's another case. And, Agree with using independent `Filler`s.